app/containers/Login/reducer.js

```diff
@@ -0,0 +1,28 @@
+/*
+ *
+ * Login reducer
+ *
+ */
+
+import { fromJS } from 'immutable';
+import { SHOW_LOGIN_MODAL, HIDE_LOGIN_MODAL } from './constants';
+
+export const initialState = fromJS({
+  content: null,
+  showModal: 'hide',
+});
+
+function loginReducer(state = initialState, action) {
+  const { type, content } = action;
+
+  switch (type) {
+    case SHOW_LOGIN_MODAL:
+      return state.set('showModal', 'show').set('content', content);
```

Comment:
Values show and hide are not very intuitive. boolean with true and false values would be better for code readability.